File: openadmin/spec/page.py
# SPDX-FileCopyrightText: 2026 OpenAdmin
#
# SPDX-License-Identifier: AGPL-3.0-or-later

from typing import List

from pydantic import BaseModel

from .component import Component
from .icons import Icon


class Page(BaseModel):
    id: str
    name: str
    description: str | None
    icon: Icon | None
    components: List[Component]

File: client/src/schemas/page.ts
// SPDX-FileCopyrightText: 2026 OpenAdmin
//
// SPDX-License-Identifier: AGPL-3.0-or-later

import { z } from "zod"
import { componentSchema } from "./component"
import { iconSchema } from "./icon"
import { colorSchema } from "./color"

export const pageSchema = z.object({
	id: z.string(),
	name: z.string(),
	description: z.string().nullable(),
	icon: iconSchema.nullable(),
	color: colorSchema.nullable(),
	components: z.array(componentSchema),
})

export type Page = z.infer<typeof pageSchema>
